polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisMetaStoreSession.java

```diff
@@ -21,32 +21,19 @@
 import jakarta.annotation.Nonnull;
 import jakarta.annotation.Nullable;
 import java.util.List;
-import java.util.function.Function;
-import java.util.function.Predicate;
 import java.util.function.Supplier;
 import org.apache.polaris.core.PolarisCallContext;
 import org.apache.polaris.core.entity.PolarisBaseEntity;
-import org.apache.polaris.core.entity.PolarisChangeTrackingVersions;
 import org.apache.polaris.core.entity.PolarisEntitiesActiveKey;
 import org.apache.polaris.core.entity.PolarisEntityActiveRecord;
 import org.apache.polaris.core.entity.PolarisEntityCore;
-import org.apache.polaris.core.entity.PolarisEntityId;
-import org.apache.polaris.core.entity.PolarisEntityType;
-import org.apache.polaris.core.entity.PolarisGrantRecord;
-import org.apache.polaris.core.entity.PolarisPrincipalSecrets;
-import org.apache.polaris.core.storage.PolarisStorageConfigurationInfo;
-import org.apache.polaris.core.storage.PolarisStorageIntegration;
 
 /**
- * Interface to the Polaris metadata store, allows to persist and retrieve all Polaris metadata like
- * metadata for Polaris entities and metadata about grants between these entities which is the
- * foundation of our role base access control model.
- *
- * <p>Note that APIs to the actual persistence store are very basic, often point read or write to
- * the underlying data store. The goal is to make it really easy to back this using databases like
- * Postgres or simpler KV store.
+ * Extends BasePersistence to express a more "transaction-oriented" control flow for backing stores
```

Comment:
Can we rename the class to something like `TransactionalMetastore` or `TransactionOrientedMetaStore`? I'd also suggest to move it to a separated package, like `org.apache.polaris.core.persistence.transactional`, so that it provides a clear structure for other types of metastore.

Reply:
Done